### PR TITLE
Bug 1648696 - Look for metrics in send_if_empty pings

### PR DESCRIPTION
**Pull request by brizental**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry to `HISTORY.rst` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

**Comment by brizental** [comment]
Hum, actually seems like this was just a problem I was having locally due to misconfiguration. I believe it might be coming from some old version of the glean parser, which I was using unintentionally.

**Comment by brizental** [comment]
Okay, yes this is an issue. I just made a big confusion where I had multiple glean_parser options installed locally, etc. Not even worht explaining here.

File: Makefile [hunk 1]
```diff
@@ -21,7 +21,7 @@ clean-build: ## remove build artifacts
 	rm -fr dist/
 	rm -fr .eggs/
 	find . -name '*.egg-info' -exec rm -fr {} +
-	find . -name '*.egg' -exec rm -f {} +
+	find . -name '*.egg' -exec rm -fr {} +
```
Comment (brizental) [inline Makefile:24]: This change is unrelated. I ran into issues while running `make clean`

File: glean_parser/markdown.py [hunk 2]
```diff
@@ -168,13 +168,14 @@ def output_markdown(
             # the description
             if isinstance(obj, pings.Ping):
                 custom_pings_cache[obj.name] = obj
+                # Pings that have `send_if_empty` set to true,
+                # might not have any metrics.
                 if obj.send_if_empty:
                     metrics_by_pings[obj.name] = []
```
Comment (brizental) [inline glean_parser/markdown.py:174]: I just realized that this might overwrite metrics that were added prior to it. I'll fix it and send a new commit.

Comment (Dexterp37) [reply glean_parser/markdown.py:174]: Cool, please flag me again once it's ready

Comment (brizental) [reply glean_parser/markdown.py:175]: It's ready!

File: tests/data/send_if_empty_with_metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+---
+$schema: moz://mozilla.org/schemas/glean/metrics/1-0-0
+
+telemetry:
+  some_metric:
+    type: uuid
+    lifetime: ping
+    description: >
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit.
```
Comment (Dexterp37) [inline tests/data/send_if_empty_with_metrics.yaml:12]: The designer part of you is now surfacing :)

**Comment by brizental** [comment]
> We should test how this will change both Fenix docs and the Glean SDK docs, if you didn't do that already.

Neither fenix or android-components have `send_if_empty` pings. Their docs should me unchanged. Glean SDK same thing, but the docs created on [this PR](https://github.com/mozilla/glean/pull/1016) will be fixed.
